packages/dcos-integration-test/extra/conftest.py

```diff
@@ -31,6 +31,13 @@ def pytest_collection_modifyitems(session, config, items):
     items[:] = new_items + last_items
 
 
+@pytest.fixture(autouse=True)
+def clean_marathon_state(dcos_api_session):
+    dcos_api_session.marathon.purge()
+    yield
+    dcos_api_session.marathon.purge()
+
```

Comment:
`autouse=True` means that this fixture is applied to  EVERYTHING, so therefore it will be "outer" to all non-autouse fixtures. If you want to ensure that an autouse fixture is inner of this, it must simply declare this as a dependent fixutre